### ISSUE TITLE
Memory Leak

### ISSUE THREAD

**Issue by matthew894** [tensorflow/tensorflow#20510]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Windows 7
- **TensorFlow installed from (source or binary)**:
Anaconda
- **TensorFlow version (use command below)**:
b'unknown' 1.8.0
CuDNN - None
GPU-Integrated
### Describe the problem
Memory Leak when running my code 

### Source code / logs
`import numpy as np`
`import os`
`import tensorflow as tf`
`tf.enable_eager_execution()`
`import tensorflow.contrib.eager as tfe`
`import csv as csv`

`import matplotlib.pyplot as plt`
`from tensorflow.contrib.opt.python.training.elastic_average_optimizer import *`
`import keras as keras`

`def parse_csv(line):`
`  example_defaults = [[0.], [0.], [0.], [0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.],[0.], [0]] # sets field types`
  
`  parsed_line = tf.decode_csv(line, example_defaults)`
  
`  # First 4 fields are features, combine into single tensor`
`  features = tf.reshape(parsed_line[:-1], shape=(50,))`
`  # Last field is the label`
`  label = tf.reshape(parsed_line[-1], shape=())`
`  return features, label`
   
`train_dataset = tf.data.TextLineDataset(fileid + "train")`
`train_dataset = train_dataset.map(parse_csv)`
`train_dataset = train_dataset.shuffle(buffer_size=1000)`
`train_dataset = train_dataset.batch(16)`

`model = tf.keras.Sequential([`
 ` tf.keras.layers.Dense(100, input_shape=(50,)),  # input shape required`
 ` tf.keras.layers.Dense(2000),`
`  tf.keras.layers.Dense(2000),`
`  tf.keras.layers.Dense(200,activation = 'sigmoid'),`
`  tf.keras.layers.Dense(200),`
 ` tf.keras.layers.Dense(2000),`
  `tf.keras.layers.Dense(200),`
 ` tf.keras.layers.Dense(2)`
  
`])`
`print (model.summary)`

`def loss(model, x, y):`
`  y_ = model(x)`
`  return tf.losses.sparse_softmax_cross_entropy(labels=y, logits=y_)`

`def grad(model, inputs, targets):`

`    with tf.GradientTape() as tape:`
    
 `       loss_value = loss(model, inputs, targets)`
  
 `   return tape.gradient(loss_value, model.variables)    `

`optimizer = tf.train.AdamOptimizer(learning_rate = 0.0010)`
`train_loss_results = []`
`train_accuracy_results = []`
`test_loss_results = []`
`test_accuracy_results = []`
`num_epochs = 2000`

`for epoch in range(num_epochs):`
`  epoch_loss_avg = tfe.metrics.Mean()`
`  epoch_accuracy = tfe.metrics.Accuracy()`
  

`  train_dataset = train_dataset.shuffle(16)`
` #  Training loop - using batches of 32`
` for x, y in train_dataset:`
`  #   Optimize the model`
  `  grads = grad(model, x, y)`
  `  optimizer.apply_gradients(zip(grads, model.variables),`
           `                   global_step=tf.train.get_or_create_global_step())`

`  #   Track progress`
`    epoch_loss_avg(loss(model, x, y))  # add current batch loss`
`     #compare predicted label to actual label`
`    epoch_accuracy(tf.argmax(model(x), axis=1, output_type=tf.int32), y)`
 
`  # end epoch`
`  train_loss_results.append(epoch_loss_avg.result())`
`  train_accuracy_results.append(epoch_accuracy.result())`
  
` if epoch % 1 == 0:
    print("Epoch {:03d}: Loss: {:.3f}, Accuracy: {:.3%}".format(epoch,
                                                                epoch_loss_avg.result(),
                                                                epoch_accuracy.result()))`
  
`  test_dataset = tf.data.TextLineDataset(fileid + "test")
  test_dataset = test_dataset.map(parse_csv)
  test_dataset = test_dataset.shuffle(buffer_size=1000)
  test_dataset = test_dataset.batch(32)`
  
 ` test_accuracy = tfe.metrics.Accuracy()`
  
 ` for (x, y) in test_dataset:
      prediction = tf.argmax(model(x), axis=1, output_type=tf.int32)
      test_accuracy(prediction, y)`
      
`  print("Test set accuracy: {:.3%}".format(test_accuracy.result()))
  train_dataset = tf.data.TextLineDataset(fileid + "train")
  train_dataset = train_dataset.map(parse_csv)
  train_dataset = train_dataset.batch(32)`
  

`del optimizer`

`test_dataset = tf.data.TextLineDataset(fileid + "test")`
`test_dataset = test_dataset.map(parse_csv)`
`test_dataset = test_dataset.shuffle(buffer_size=1000)`
`test_dataset = test_dataset.batch(32)`

`test_accuracy = tfe.metrics.Accuracy()`

`for (x, y) in test_dataset:`
 ` prediction = tf.argmax(model(x), axis=1, output_type=tf.int32)`
`  test_accuracy(prediction, y)`

`print("Test set accuracy: {:.3%}".format(test_accuracy.result()))`

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by matthew894** [comment]
Done

**Comment by tensorflowbutler** [comment]
Nagging Assignee @skye: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by skye** [comment]
Can you fix the formatting of your code? It's very hard to read right now... you can create code blocks using ```, see https://help.github.com/articles/creating-and-highlighting-code-blocks/

**Comment by matthew894** [comment]
done

**Comment by skye** [comment]
How do you know there's a memory leak? Can you try to narrow down where in your code the leak occurs?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @skye: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @skye: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by harshini-gadige** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
